mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -288,24 +280,47 @@ protected void onCreate(Bundle savedInstanceState) {
         Util.setActivityTheme(this);
         super.onCreate(savedInstanceState);
 
+        String controllerClassName = getResources().getString(R.string.controller_class);
+        try {
+            Class<?> controllerClass = Class.forName(controllerClassName);
+            Constructor<?> constructor = controllerClass.getConstructor(OpenHABMainActivity.class);
+            mController = (ActivityController) constructor.newInstance(this);
+        } catch (Exception e) {
+            Log.wtf(TAG, "Could not instantiate activity controller class '"
+                    + controllerClassName + "'");
+            throw new RuntimeException(e);
+        }
+
         setContentView(R.layout.activity_main);
 
+        // inflate the controller dependent content view
+        ViewStub contentStub = findViewById(R.id.content_stub);
+        contentStub.setLayoutResource(mController.getContentLayoutResource());
+        mController.initViews(contentStub.inflate());
+
         setupToolbar();
         setupDrawer();
         gcmRegisterBackground();
-        setupPager();
 
         mViewPool = new RecyclerView.RecycledViewPool();
         MemorizingTrustManager.setResponder(this);
 
         // Check if we have openHAB page url in saved instance state?
         if (savedInstanceState != null) {
             openHABBaseUrl = savedInstanceState.getString("openHABBaseUrl");
-            sitemapRootUrl = savedInstanceState.getString("sitemapRootUrl");
-            mStartedWithNetworkConnectivityInfo = savedInstanceState.getParcelable("startedWithNetworkConnectivityInfo");
+            mLastKnownNetworkInfo = savedInstanceState.getParcelable("lastNetworkInfo");
```

Comment:
Constants for "lastNetworkInfo" and others?

Reply:
Yeah, sure, but maybe leave it to the separate cleanup work to do it consistently everywhere?